numpy/_core/src/multiarray/dtypemeta.c

```diff
@@ -1387,3 +1387,22 @@ PyArray_DTypeMeta *_Datetime_dtype = NULL;
 PyArray_DTypeMeta *_Timedelta_dtype = NULL;
 PyArray_DTypeMeta *_Object_dtype = NULL;
 PyArray_DTypeMeta *_Void_dtype = NULL;
+
+
+
+/*NUMPY_API
+ * Fetch the ArrFuncs struct which new lives on the DType and not the
+ * descriptor.  Use of this struct should be avoided but remains necessary
+ * for certain functionality.
+ *
+ * The use of any slot besides getitem, setitem, copyswap, and copyswapn
+ * is only valid after checking for NULL.  Checking for NULL is generally
+ * encouraged.
+ *
+ * The public name of this is a macro until 1.x BC is irrelevant.
```

Comment:
This helps but I'm still a little confused why we need to publicly expose a static inline function which calls thing function which calls another static inline function. I guess it's just dealing with naming conflicts but it would be nice to explicitly spell out the problem and maybe expand this comment a bit.

Reply:
1. Internally, I thought an inline function is nicer for the simple pointer chasing that is going on.
2. Publically, an inline function isn't possible because the struct is fully opaque.  So,
3. the public static inline is then needed allow defining it in a backcompat way.

Reply:
I added a few sentences here to try to clarify it.  I agree it is an unfortunate dance, necessary due to trying to hide it fully, but also needing the backcompat.